### DISCUSSION TITLE
Need Best practices example setting up semantic kernel with Microsoft Bot Framework

### DISCUSSION THREAD

**Discussion by albin-98** [microsoft/semantic-kernel#8361] [General]
Need Best practices example, setting up multi-agent semantic kernel with Microsoft Bot Framework

**Comment by albin-98** [comment]
I've built a chatbot using the **Microsoft Bot Framework**, which handles various use cases. Some of these use cases require **Azure OpenAI**, while others do not. To manage this efficiently, I've decided to integrate Semantic Kernel as a separate class library. My main objective is to **implement a RAG operation** using a semantic kernel within the chatbot. Can anyone help me suggest a good method?

**Comment by sophialagerkranspandey** [comment]
What is your expected scenario - are there multiple agents talking to each other via bot framework or within a single bot are there multiple  agents in the bot.

**Comment by albin-98** [comment]
I have a chatbot with many use cases that utilizes Cognitive Language Understanding (CLU) for intent identification. 

I would like to know if the kernel can be employed in the orchestrator layer and whether it is possible to invoke existing dialogs from the kernel ?

 Is it feasible to call multiple agents through a single bot?
